### ISSUE TITLE
Managed identity connection string is not secure:  `data_factory_linked_service_azure_sql_database_resource`

### ISSUE THREAD

**Issue by drjwelch** [terraform-providers/terraform-provider-azurerm#12298]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

terraform 0.14.7
azurerm 2.64

### Affected Resource(s)

* `data_factory_linked_service_azure_sql_database_resource`

### Terraform Configuration Files

```hcl
resource "azurerm_data_factory_linked_service_azure_sql_database" "example" {
  name                  = "example"                  
  resource_group_name   = azurerm_resource_group.myRg.name
  data_factory_name     = azurerm_data_factory.myFactory.name
  connection_string     = "Data Source=${azurerm_sql_server.myServer.fully_qualified_domain_name};Initial Catalog=${azurerm_sql_database.myDb.name};Connection Timeout=30;Integrated Security=False;Encrypt=True"
  integration_runtime_name = azurerm_data_factory_integration_runtime_self_hosted.myIRT.name
  use_managed_identity  = true
}
```
### Expected Behaviour
The data factory displays the linked service as using Managed Identity to connect
The data factory linked service resource JSON can be imported into a git repository

### Actual Behaviour
The data factory displays the linked service as using SQL Authentication to connect
The data factory linked service resource JSON cannot be imported into a git repository because the connection string was pushed into the API as a secure string and therefore cannot be read in the factory UI

### Steps to Reproduce

1. `terraform apply`
2.  Open data factory UI and open the linked service / browse to its JSON code

### Important Factoids

In the [Azure resource](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/datafactory/data_factory_linked_service_azure_sql_database_resource.go) line 201:

```
if v, ok := d.GetOk("connection_string"); ok {
        sqlDatabaseProperties.ConnectionString = &amp;datafactory.SecureString{
        	Value: utils.String(v.(string)),
        	Type:  datafactory.TypeSecureString,
        }
```
which always passes the connection string to the API as a SecureString.

When the resource is using managed identity, this could be passed as String instead ([docs](https://docs.microsoft.com/en-us/rest/api/datafactory/linked-services/create-or-update#azuresqldatabaselinkedservice)) because there are no secrets in the connection string when using managed identity.

### References

Related to this [issue](https://github.com/terraform-providers/terraform-provider-azurerm/issues/10758), possibly

**Comment by drjwelch** [comment]
@favoretti I will try to change this myself as per your help on #12180, if you would be willing to consider the PR when I have done it.  Thank you!

**Comment by favoretti** [comment]
@drjwelch Basically just checkout master, pull it from origin to synchronise your fork, create a new branch and do it, those changes aren't in the way of each other, so they shouldn't result in merge conflicts. :)

**Comment by drjwelch** [comment]
Thanks for your advice and support.  After committing the changes on master by mistake I think I rolled them back and committed again on the branch for this PR (https://github.com/terraform-providers/terraform-provider-azurerm/pull/12359) - however the branch also has the changes from the previous issue 12180 :(
Sorry for being a noob and hope this can still be useful!

**Comment by katbyte** [comment]
closed by #12359
